packages/terra-button/src/Button.jsx

```diff
@@ -6,105 +6,238 @@ import styles from './Button.scss';
 
 const cx = classNames.bind(styles);
 
+const KEYCODES = {
+  ENTER: 13,
+  SPACE: 32,
+  TAB: 9,
+};
+
+const ButtonVariants = {
+  NEUTRAL: 'neutral',
+  EMPHASIS: 'emphasis',
+  'DE-EMPSHASIS': 'de-emphasis',
+  ACTION: 'action',
+  UTILITY: 'utility',
+};
+
+const ButtonTypes = {
+  BUTTON: 'button',
+  SUBMIT: 'submit',
+  RESET: 'reset',
+};
+
 const propTypes = {
   /**
-   * Child Nodes
+   * Child nodes.
    */
   children: PropTypes.node,
   /**
-   * Sets the href. When set will render the component as an anchor tag
+   * Sets the href. When set will render the component as an anchor tag.
    */
   href: PropTypes.string,
   /**
-   * An optional icon. Nested inline with the text when provided
+   * An optional icon. Nested inline with the text when provided.
    */
   icon: PropTypes.element,
   /**
-   * Whether or not the button should display as a block
+   * Whether or not the button should only display as an icon.
    */
-  isBlock: PropTypes.bool,
+  isIconOnly: PropTypes.bool,
   /**
-   * Whether or not the button has reduced padding
+   * Whether or not the button should display as a block.
    */
-  isCompact: PropTypes.bool,
+  isBlock: PropTypes.bool,
   /**
-   * Whether or not the button should be disabled
+   * Whether or not the button should be disabled.
    */
   isDisabled: PropTypes.bool,
   /**
-   * Reverses the position of the icon and text
+   * Reverses the position of the icon and text.
    */
   isReversed: PropTypes.bool,
   /**
-   * Callback function triggered when clicked
+   * Callback function triggered when clicked.
    */
   onClick: PropTypes.func,
   /**
-   * Sets the button size. One of `tiny`, `small`, `medium`, `large`, `huge`
+   * Callback function triggered when button loses focus.
    */
-  size: PropTypes.oneOf(['tiny', 'small', 'medium', 'large', 'huge']),
+  onBlur: PropTypes.func,
   /**
-   * Sets the button text
+   * Callback function triggered when button gains focus.
    */
-  text: PropTypes.string,
+  onFocus: PropTypes.func,
   /**
-   * Sets the button type. One of `button`, `submit`, or `reset`
+   * Callback function triggered when key is pressed.
    */
-  type: PropTypes.oneOf(['button', 'submit', 'reset']),
+  onKeyDown: PropTypes.func,
   /**
-   * Sets the button variant. One of `primary`, `secondary`, or `link`
+   * Callback function triggered when key is released.
    */
-  variant: PropTypes.oneOf(['default', 'link', 'primary', 'secondary']),
+  onKeyUp: PropTypes.func,
+  /**
+   *  React `ref` callback to receive and store a reference to the mounted button instance.
+   */
+  refCallback: PropTypes.func,
```

Comment:
We shouldn't need this prop. Making the Button a class should enable the ref callback automatically